named_arrays/_scalars/scalars.py

```diff
@@ -819,6 +826,13 @@ def from_scalar_array(
 
         return self
 
+    @classmethod
+    def from_xarray(cls, dataarray: xr.DataArray) -> Self:
+        """
+        Convert xr.DataArray into a ScalarArray
```

Comment:
Same here, I think you'll also need to add xarray to intersphinx in `docs/conf.py`